Added TXT functionality for both Android & iOS

dotemacs opened the pull request:
Hello

This PR implements the TXT record for both Android & iOS.

I've implemented the functionality for iOS and pulled in the changes from #99 as were implemented by @FrankFundel. I also added the missing bits as requested in that PR:

- default parameter for TXT argument (please double check this, if you're happy with this approach)
- updated the README documentation for `publishService`

But unlike in #99, I didn't touch `package.json`.

Please see what you make of this change and let me know if something else should be added.

Also thanks for the package @balthazar.

balthazar commented:
Looking good, this won't crash on Android

Do we really need the `if (txt)` in iOS though now that you passed an empty object by default? You should either remove it or do the check if the txt dictionary is empty so we don't have to call `setTXTRecordData:txtData` if it's empty (might be good to do this check on android too, but it's just micro optimization

dotemacs commented:
> On 20 Apr 2020, at 4:24 pm, Balthazar Gronon <notifications@github.com> wrote:
> 
> ﻿
> Looking good, this won't crash on Android
> 
> Do we really need the if (txt) in iOS though now that you passed an empty object by default?
> 
I think that we do.

The reason I added the if statement was because when I omitted the `txt` argument to the JavaScript function, the ObjC method would complain about the nil value being passed in. 
> You should either remove it or do the check if the txt dictionary is empty so we don't have to call setTXTRecordData:txtData if it's empty
> 

This is exactly the reason why I added it. Because my understanding is that this is idiomatic ObjC way doing exactly what you’re describing. 

See:  https://nshipster.com/nil/ 

Happy to be proved wrong.
> (might be good to do this check on android too, but it's just micro optimization
> 

I’d rather leave it as is, because, just as you say it’s micro...

Thanks
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

balthazar commented:
Alrighty, thank you!
